src/Controller/EnMarche/MembershipController.php

```diff
@@ -31,18 +32,20 @@ class MembershipController extends Controller
      * @Route("/inscription", name="app_membership_register")
      * @Method("GET|POST")
      */
-    public function registerAction(Request $request): Response
+    public function registerAction(Request $request, GeoCoder $geoCoder): Response
     {
-        $membership = MembershipRequest::createWithCaptcha($request->request->get('g-recaptcha-response'));
-        $form = $this->createForm(MembershipRequestType::class, $membership)
-            ->add('submit', SubmitType::class, ['label' => 'J\'adhère'])
-        ;
+        $membership = MembershipRequest::createWithCaptcha(
+            $geoCoder->getCountryCodeFromIp($request->getClientIp()),
+            $request->request->get('g-recaptcha-response')
+        );
+
+        $form = $this->createForm(NewMemberShipRequestType::class, $membership);
 
         try {
             if ($form->handleRequest($request)->isSubmitted() && $form->isValid()) {
                 $this->get('app.membership_request_handler')->handle($membership);
 
-                return $this->redirectToRoute('app_membership_donate');
```

Comment:
Is it intended to break the donation on boarding flow here?

Reply:
Yes a new one has been decided ;-)